devito/arguments.py

```diff
@@ -77,14 +77,31 @@ class ScalarArgument(Argument):
     def __init__(self, name, provider, reducer=lambda old, new: new, default_value=None):
         super(ScalarArgument, self).__init__(name, provider, default_value)
         self.reducer = reducer
+        self._frozen = False
+
+    def reset(self):
+        super(ScalarArgument, self).reset()
+        self._frozen = False
```

Comment:
what's ``frozen`` for?

Reply:
Added docstring to `verify` for explanation